libr/include/r_types.h

```diff
@@ -654,4 +654,16 @@ static inline void r_run_call10(void *fcn, void *arg1, void *arg2, void *arg3, v
 # endif
 #endif
 
+// reference counter
+typedef void (*RFreeCallback)(void *a);
+typedef struct r_ref_t {
+	volatile size_t count;
+	RFreeCallback free;
+} RRef;
+
+#define r_ref_new(x) RRef{0,x}
```

Comment:
`r_ref_init` maybe? it does not create a new object, so the name is misleading. Also, I think you'd want to embed `RRef` in other structs.

Reply:
done